src/fhir/resources.ts

```diff
@@ -31,7 +59,343 @@ export const createPatientEntry = (form: typeof patientSchema) => {
       }]
     }];
   patient.birthDate = form.dateOfBirth;
-  patient.text = {status: "generated", div: `${form.firstName} ${form.lastName}`};
+  patient.text = generatedNarrative(form.firstName, form.lastName);
   patient.resourceType = "Patient";
-  return patient;
+  patient.managingOrganization = reference("Organization", organisationId);
+  return {identifier: form.mrn, id: patient.id, resource: patient};
 };
+
+export const organisationAndId = (form: typeof addressSchema): ResourceAndIds => {
+  const org = new Organization();
+  org.id = uuidv4();
+  org.identifier = [{value: GOSH_GENETICS_IDENTIFIER}];
+  org.resourceType = "Organization";
+  org.active = true;
+  org.type = [{
+    "coding": [{
+      "system": "http://terminology.hl7.org/CodeSystem/organization-type",
+      "code": "prov",
+      "display": "Healthcare Provider"
+    }],
+    "text": "Healthcare Provider"
+  }];
+  org.name = form.name;
+  org.address = [{
+    line: form.streetAddress,
+    city: form.city,
+    postalCode: form.postalCode,
+    country: form.country,
+  }];
+  org.text = generatedNarrative(form.name);
+
+  return {identifier: GOSH_GENETICS_IDENTIFIER, id: org.id, resource: org};
+}
+
+export const practitionersAndIds = (result: typeof reportDetailSchema) => {
+  return {
+    authoriser: practitionerAndId(result.authorisingScientist, result.authorisingScientistTitle),
+    reporter: practitionerAndId(result.reportingScientist, result.reportingScientistTitle),
+  }
+}
+
+/**
+ * Create a practitioner from limited data.
+ * @param fullName the first word will be the firstName, remaining words will be lastName
+ * @param title role or job title within the laboratory
+ */
+const practitionerAndId = (fullName: string, title: string): ResourceAndIds => {
+  const nameSplit = fullName.split(/\s/g);
+  const firstName = nameSplit[0];
+  const lastName = nameSplit.slice(1).join(" ");
+
+  const practitioner = new Practitioner();
+  practitioner.id = uuidv4();
```

Comment:
oh is there, fun! Happy to switch to that to keep dependencies lighter if you think it's worth consolidating on that?

Reply:
I think it's fine to use uuid here. The useId hook is fairly new and it would require refactoring your files into functional components. So might be more trouble than its worth at this stage.